tools/shared.py

```diff
@@ -273,14 +274,42 @@ def generate_config(path, first_time=False):
 JAVA = None
 PYTHON = None
 JS_ENGINE = None
+JS_ENGINES = []
 COMPILER_OPTS = []
 
-try:
+
+def parse_config_file():
+  """Parse the emscripten config file using python's exec"""
+  config = {}
   config_text = open(CONFIG_FILE, 'r').read() if CONFIG_FILE else EM_CONFIG
-  exec(config_text)
-except Exception as e:
-  logging.error('Error in evaluating %s (at %s): %s, text: %s' % (EM_CONFIG, CONFIG_FILE, str(e), config_text))
-  sys.exit(1)
+  try:
+    exec(config_text, config)
+  except Exception as e:
+    logging.error('Error in evaluating %s (at %s): %s, text: %s' % (EM_CONFIG, CONFIG_FILE, str(e), config_text))
+    sys.exit(1)
+
+  CONFIG_KEYS = (
```

Comment:
I believe `EMSCRIPTEN_NATIVE_OPTIMIZER` can also be loaded from the config file, see

https://github.com/kripken/emscripten/blob/incoming/tools/js_optimizer.py#L80